docs/src/extend/custom-rule-tutorial.md

```diff
@@ -0,0 +1,452 @@
+
+The `RuleTester#run()` method tests the rule against valid and invalid test cases. If the rule fails to pass any of the test scenarios, this method throws an error.
+`RuleTester` requires that at least one valid and one invalid test scenarios be present.
+
+```javascript
+// enforce-foo-bar.test.js
+const {RuleTester} = require("eslint");
+const fooBarRule = require("./enforce-foo-bar");
+
+const ruleTester = new RuleTester({
+  // Must use at least ecmaVersion 2015 because
+  // that's when `const` variable introduced.
+  parserOptions: { ecmaVersion: 2015 }
+});
+
+// Throws error if the tests in ruleTester.run() do not pass
+ruleTester.run(
+  "foo-bar", // rule name
+  fooBarRule, // rule code
+  { // checks
+    // 'valid' checks cases that should pass
+    valid: [{
+      code: "const foo = 'bar';",
+    }],
+    // 'invalid' checks cases that should not pass
+    invalid: [{
+      code:"const foo = 'baz';",
+      output: 'const foo = "bar";',
+      errors: 1,
+    }],
+  }
+);
+
+console.log("All tests passed!");
+```
+
+## Step 7: Bundle the Custom Rule in a Plugin
+
+Now that you've written the custom rule and validated that it works, you can include it in a plugin. Using a plugin, you can share the rule in an npm package to use in other projects.
+
+Create the file for the plugin:
+
+```shell
+touch eslint-plugin-example.js
+```
+
+And now write the plugin code. Plugins are just exported JavaScript objects. To include a rule in a plugin, include it in the plugin's `rules` object, which contains key-value pairs of rule names and their source code.
+
+To learn more about creating plugins, refer to [Create Plugins](plugins).
+
+```javascript
+// eslint-plugin-example.js
+
+const fooBarRule = require("./enforce-foo-bar");
+const plugin = { rules: { "foo-bar": fooBarRule } };
+module.exports = plugin;
+```
+
+## Step 8: Use the Plugin Locally
+
+You can use a locally defined plugin to execute the custom rule in your project. To use a local plugin, specify the path to the plugin in the `plugins` property of your ESLint configuration file.
+
+You might want to use a locally defined plugin in one of the following scenarios:
+
+* You want to test the plugin before publishing it to npm.
+* You want to use a plugin, but do not want to publish it to npm.
+
+Before you can add the plugin to the project, create an ESLint configuration for your project using a [flat configuration file](../use/configure/configuration-files-new), `eslint.config.js`:
+
+```shell
+touch eslint.config.js
+```
+
+Then, add the following code to `eslint.config.js`:
+
+```javascript
+// eslint.config.js
+"use strict";
+
+// Import the ESLint plugin locally
+const eslintPluginExample = require("./eslint-plugin-example");
+
+module.exports = [
+    {
+        files: ["**/*.js"],
+        languageOptions: {
+            sourceType: "commonjs",
+            ecmaVersion: "latest",
+        },
+        // Using the eslint-plugin-example plugin downloaded from npm
```

Comment:
This configuration uses local file `./eslint-plugin-example.js`, not a plugin downloaded from npm.

Reply:
```suggestion
        // Using the eslint-plugin-example plugin defined locally
```